src/components/todo/TodoForm.tsx

```diff
@@ -71,27 +109,72 @@ const TodoForm = ({
     }
   };
 
+  useEffect(() => {
+    if (editTask) {
+      setValue("title", editTask.title);
+      setValue("dueDate", moment(editTask.dueDate).format("YYYY-MM-DD"));
+      setValue("category", editTask.category);
+    }
+  }, [editTask]);
+
   return (
     <>
-      <form className="todo-form" onSubmit={handleSubmit(handleAdd)}>
-        <input
-          {...register("title", { required: true })}
-          placeholder="Please enter todo"
-        />
-        <span className="error">
-          {errors.title?.type === "required" && "Please enter todo"}
-        </span>
-        <div className="input-date-container">
+      <form
+        className="todo-form"
+        onSubmit={handleSubmit(editTask ? updateTaskHandeler : handleAdd)}
+      >
+        <div className="task-input-group">
+          <div className="task-input">
+            <input
+              {...register("title", { required: true })}
+              placeholder="Please enter todo"
+            />
+
+            <span className="error">
+              {errors.title?.type === "required" && "Please enter todo"}
+            </span>
+          </div>
           <input
             {...register("dueDate", { required: true, valueAsDate: true })}
             defaultValue={moment(new Date()).format("YYYY-MM-DD")}
             placeholder="Please enter date of todo"
             className="input-date"
             type="date"
           />
-          <span className="error">
-            {errors.dueDate?.type === "required" && "Please enter a date"}
-          </span>
+        </div>
+        <div className="input-category-group">
+          <div className="input-set">
+            <label htmlFor="">Category</label>
+            <select {...register("category")}>
+              {categories.map((category, id) => (
+                <option
+                  key={category.id}
+                  defaultValue={
+                    category.id === editTask?.category ? category.id : 0
+                  }
+                  value={category.id}
+                >
+                  {category.title}
+                </option>
+              ))}
+            </select>
```

Comment:
We can create a generic component for select dropdown

Reply:
Prepared generic component for select and input